src/unicorn/mod.rs

```diff
@@ -12,6 +12,7 @@ use std::rc::Rc;
 pub mod bitblasting;
 pub mod bitblasting_dimacs;
 pub mod bitblasting_printer;
+pub mod boolector_impl;
```

Comment:
I think if we guard this with `#[cfg(feature = "boolector")]` then the boolector_impl.rs file will only be compiled when this feature is active. This means the boolector_impl.rs file can be simplified to not contain any feature checks at all. Can you check whether this indeed works?